CMakeLists.txt

```diff
@@ -84,6 +85,13 @@ if(THREADCHECK)
     string(APPEND CMAKE_CXX_FLAGS " -fsanitize=thread")
     message("-- Thread Sanitizer Enabled")
 endif()
+if(CODECOVERAGE)
+    string(APPEND CMAKE_CXX_FLAGS " -g -O0")
```

Comment:
1) Shouldnt this be moved to Makefile like how we overwrite default for asan/tsan report generation.
2)Also wouldn't code coverage be generated with default flags?

Reply:
optimisation flag have been moved to Makefile like asan/tsan.